### PR TITLE
Fix collective APIs cannot be recognized when building docs

### PR DESCRIPTION
**Pull request by HermitSun**
### PR types

Bug fixes
### PR changes

APIs
### Describe

Fix some APIs in `paddle.distributed` cannot be recognized when building docs.

### CODE REVIEW

File: python/paddle/distributed/__init__.py [hunk 1]
```diff
@@ -51,7 +51,7 @@
 from .collective import P2POp  # noqa: F401
 from .collective import reduce_scatter  # noqa: F401
 
-from .communication import *  # noqa: F401
+from .communication import stream  # noqa: F401
```
Comment (LiYuRio) [inline python/paddle/distributed/__init__.py]: 这里是不是不用修改了

Comment (HermitSun) [reply python/paddle/distributed/__init__.py]: 是，我只是觉得明确一下会比较好，要改回去吗

File: python/paddle/distributed/__init__.py [hunk 2]
```diff
@@ -76,7 +76,7 @@
     "gloo_release", "QueueDataset", "split", "CountFilterEntry",
     "ShowClickEntry", "get_world_size", "get_group", "all_gather",
     "all_gather_object", "InMemoryDataset", "barrier", "all_reduce", "alltoall",
-    "send", "reduce", "recv", "ReduceOp", "wait", "get_rank",
-    "ProbabilityEntry", "ParallelMode", "is_initialized", "isend", "irecv",
-    "reduce_scatter", "rpc"
+    "alltoall_single", "send", "reduce", "recv", "ReduceOp", "wait", "get_rank",
+    "ProbabilityEntry", "ParallelMode", "is_initialized",
+    "destroy_process_group", "isend", "irecv", "reduce_scatter", "rpc", "stream"
```
Comment (LiYuRio) [inline python/paddle/distributed/__init__.py:81]: 这里的stream是必要的吗

File: python/paddle/distributed/__init__.py (hunk 1 above)
Comment (HermitSun) [reply python/paddle/distributed/__init__.py]: done

**Review by LiYuRio — APPROVED** [review]
lgtm

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by liuTINA0907 — APPROVED** [review]

**Pull request merged**